crates/matrix-sdk-base/src/rooms/members.rs

```diff
@@ -13,28 +13,188 @@
 // limitations under the License.
 
 use std::{
-    collections::{BTreeSet, HashMap},
+    collections::{BTreeMap, BTreeSet, HashMap},
+    mem,
     sync::Arc,
 };
 
+use bitflags::bitflags;
 use ruma::{
     events::{
+        ignored_user_list::IgnoredUserListEventContent,
         presence::PresenceEvent,
         room::{
-            member::MembershipState,
+            member::{MembershipState, RoomMemberEventContent},
             power_levels::{PowerLevelAction, RoomPowerLevels, RoomPowerLevelsEventContent},
         },
         MessageLikeEventType, StateEventType,
     },
     MxcUri, OwnedUserId, UserId,
 };
+use tracing::debug;
 
+use super::Room;
 use crate::{
     deserialized_responses::{DisplayName, MemberEvent, SyncOrStrippedState},
-    store::ambiguity_map::is_display_name_ambiguous,
+    store::{ambiguity_map::is_display_name_ambiguous, Result as StoreResult, StateStoreExt},
     MinimalRoomMemberEvent,
 };
 
+impl Room {
```

Comment:
It would probably make sense to create an empty API struct like we have for `Client::encryption()` for this and some of the others, that way we wouldn't have `impl Room` blocks scattered around and the docs would then be more consumable as well.

I.e. if you want to know about `members` methods on the `Room` you would look at whichever struct we add here.

Though this might be something for a separate PR and would be a pretty big breaking change.